include/alpaka/workdiv/WorkDivHelpers.hpp

```diff
@@ -315,6 +357,79 @@ namespace alpaka
                 getExtents(gridElemExtent),
                 getExtents(threadElemExtents),
                 getAccDevProps<TAcc>(dev),
+                static_cast<Idx<TAcc>>(0u),
+                blockThreadMustDivideGridThreadExtent,
+                gridBlockExtentSubDivRestrictions);
+        using V [[maybe_unused]] = Vec<Dim<TGridElemExtent>, Idx<TGridElemExtent>>;
+        ALPAKA_UNREACHABLE(WorkDivMembers<Dim<TGridElemExtent>, Idx<TGridElemExtent>>{V{}, V{}, V{}});
+    }
+
+    //! \tparam TDev The type of the device.
+    //! \tparam TKernelBundle The type of the bundle of kernel and the arguments. Kernel is used to get number of
+    //! threads per block, this number could be less than or equal to the number of threads per block according to
+    //! device properties.
+    //! \tparam TGridElemExtent The type of the grid element extent.
+    //! \tparam TThreadElemExtent The type of the thread element extent.
+    //! \param dev The device the work division should be valid for.
+    //! \param kernelBundle An instance of a class consisting Kernel function and its arguments
+    //! \param gridElemExtent The full extent of elements in the grid.
+    //! \param threadElemExtents the number of elements computed per thread.
+    //! \param blockThreadMustDivideGridThreadExtent If this is true, the grid thread extent will be multiples of the
+    //! corresponding block thread extent.
+    //!     NOTE: If this is true and gridThreadExtent is prime (or otherwise bad chosen) in a dimension, the block
+    //!     thread extent will be one in this dimension.
+    //! \param gridBlockExtentSubDivRestrictions The grid block extent subdivision restrictions.
+    //! \return The work division.
+    template<
+        typename TAcc,
+        typename TDev,
+        typename TKernelBundle,
+        typename TGridElemExtent = Vec<Dim<TAcc>, Idx<TAcc>>,
+        typename TThreadElemExtent = Vec<Dim<TAcc>, Idx<TAcc>>>
+    ALPAKA_FN_HOST auto getValidWorkDivForKernel(
+        [[maybe_unused]] TDev const& dev,
+        TKernelBundle const& kernelBundle,
+        [[maybe_unused]] TGridElemExtent const& gridElemExtent = Vec<Dim<TAcc>, Idx<TAcc>>::ones(),
+        [[maybe_unused]] TThreadElemExtent const& threadElemExtents = Vec<Dim<TAcc>, Idx<TAcc>>::ones(),
+        [[maybe_unused]] bool blockThreadMustDivideGridThreadExtent = true,
+        [[maybe_unused]] GridBlockExtentSubDivRestrictions gridBlockExtentSubDivRestrictions
+        = GridBlockExtentSubDivRestrictions::Unrestricted)
+        -> WorkDivMembers<Dim<TGridElemExtent>, Idx<TGridElemExtent>>
```

Comment:
maybe it is better to use 

```C++
template<
        typename TAcc,
        typename TDev,
        typename TKernelBundle,
        typename TGridElemExtent = Vec<Dim<TAcc>, Idx<TAcc>>,
        typename TThreadElemExtent = Vec<Dim<TAcc>, Idx<TAcc>>>
    ALPAKA_FN_HOST auto getValidWorkDivForKernel(
        [[maybe_unused]] TDev const& dev,
        TKernelBundle const& kernelBundle,
        [[maybe_unused]] TGridElemExtent const& gridElemExtent = Vec<Dim<TAcc>, Idx<TAcc>>::ones(),
        [[maybe_unused]] TThreadElemExtent const& threadElemExtents = Vec<Dim<TAcc>, Idx<TAcc>>::ones(),
        [[maybe_unused]] bool blockThreadMustDivideGridThreadExtent = true,
        [[maybe_unused]] GridBlockExtentSubDivRestrictions gridBlockExtentSubDivRestrictions
        = GridBlockExtentSubDivRestrictions::Unrestricted)
        -> std::endbale_if_t<Dim<TGridElemExtent>::value == Dim<Acc>::value && Dim<TThreadElemExtent>::value == Dim<Acc>::value && std::is_same_v<Idx<TGridElemExtent>, Idx<Acc>> && std::is_same_v<Idx<TThreadElemExtent>, Idx<Acc>>,WorkDivMembers<Dim<TGridElemExtent>, Idx<TGridElemExtent>>>
```

instead of having aserts within the function.